.github/workflows/pr_test.yml

```
@@ -0,0 +1,31 @@
+name: PR Testing
+
+on:
+  workflow_call:
+# In the repo that uses this workflow, this should be set to
+# on:
+#   - pull_request 
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        puppet_versions:
+          - '7'
+          - '8'
+
+    name: Puppet ${{ matrix.puppet_versions }}
+    steps:
+    - uses: actions/checkout@v1
```

I think that v1 and v2 are deprecated
```suggestion
    - uses: actions/checkout@v3
```